aws-datastore/src/androidTest/java/com/amplifyframework/datastore/MultiAuthSyncEngineInstrumentationTest.java

```diff
@@ -0,0 +1,1036 @@
+        DataStoreCategory dataStoreCategory = new DataStoreCategory();
+        dataStoreCategory.addPlugin(awsDataStorePlugin);
+        dataStoreCategory.configure(dataStoreCategoryConfiguration, context);
+        dataStoreCategory.initialize(context);
+        dataStore = SynchronousDataStore.delegatingTo(dataStoreCategory);
+    }
+
+    // The following link was helpful in finding the right setup
+    // https://github.com/googleapis/google-auth-library-java#google-auth-library-oauth2-http
+    private void oidcLogin() {
+        try {
+            IdToken idToken = googleServiceAccount.idTokenWithAudience(AUDIENCE, Collections.emptyList());
+            token.set(idToken.getTokenValue());
+        } catch (IOException exception) {
+            LOG.warn("An error occurred while trying to authenticate against OIDC provider", exception);
+        }
+    }
+
+    /**
+     * Test tear-down activities.
+     */
+    @AfterClass
+    public static void cleanupAfterAllTests() {
+        Log.i("TearDown", "Cleaning up databases");
+        for (String db : getApplicationContext().databaseList()) {
+            Log.i("TearDown", "Removing " + db);
+            getApplicationContext().deleteDatabase(db);
+            Log.i("TearDown", db + " removed");
+        }
+        Log.i("TearDown", "Cleanup completed.");
+    }
+
+    private void verifyScenario(Class<? extends Model> modelType,
+                                  boolean signInToCognito,
+                                  boolean signInWithOidc,
+                                  AuthorizationType expectedAuthType) throws AmplifyException, IOException {
+        configure(modelType, signInToCognito, signInWithOidc, expectedAuthType);
+        String modelId = UUID.randomUUID().toString();
+        Model testRecord = createRecord(modelType, modelId);
+        HubAccumulator expectedEventAccumulator = null;
+        if (expectedAuthType != null) {
+            expectedEventAccumulator =
+                HubAccumulator
+                    .create(HubChannel.DATASTORE, publicationOf(modelType.getSimpleName(), testRecord.getId()), 1)
+                    .start();
+        } else {
+            expectedEventAccumulator = HubAccumulator
+                    .create(HubChannel.DATASTORE, networkStatusFailure(), 1)
+                    .start();
+        }
+        dataStore.start();
+        dataStore.save(testRecord);
+        expectedEventAccumulator.await(TIMEOUT_SECONDS, TimeUnit.SECONDS);
+        assertFalse(requestInterceptor.hasUnexpectedRequests());
+        // Sign the user out if sign-in was required.
+        if (signInToCognito) {
+            auth.signOut(AuthSignOutOptions.builder().build());
+        }
+    }
+
+    /**
+     * Create a instance of the model using the private constructor via reflection.
+     * @return A instance of the model being tested.
+     */
+    private Model createRecord(Class<?> modelType, String modelId) {
```

Comment:
we can probably make this more strongly typed with something like
```suggestion
    private <T extends Model> T createRecord(Class<T> modelType, String modelId) {
```